---
title: Add assay data
sidebar:
  order: 4
authors:
  - dominik-brilhaus
  - lukas-weil
---

import ViolaSays from '@components/ViolaSays.astro'
import { Steps } from '@astrojs/starlight/components';
import { Card } from '@astrojs/starlight/components';
import { FileTree } from '@astrojs/starlight/components';


In **assays** you can store data and metadata from measurements.

![](@images/start-here/arc-prototypic-assay-identify.svg)

## Assay 1: Sugar Measurement

![](@images/start-here/arc-prototypic-assay-sugar.svg)


### Add a new assay

<Card icon="pen" title="ARCitect">
    <Steps>

    1. Click on the plus icon next to `assays` to add a new assay.
    2. Enter a name for the `New Assay`.

        <ViolaSays>
            SugarMeasurement
        </ViolaSays>

    3. Click `New Assay`.
    4. ARCitect adds the assay `SugarMeasurement` including the folders `dataset` and `protocols` as well as a `README.md` and the `isa.assay.xlsx` workbook.
    
        <FileTree>
        - assays
            - **SugarMeasurement**
                - dataset
                    - ...
                - protocols
                    - ...
                - isa.assay.xlsx
                - README.md
            - ...
        </FileTree>

        :::note
        The `isa.assay.xlsx` is not displayed inside ARCitect.
        :::

    5. Click on the assay's name (`SugarMeasurement`) in the file tree to edit the assay metadata in the right panel.
    6. Here you can add metadata about your assay.

        <ViolaSays>
            Ok, so my **measurement type** is "sugar measurement". 
            And to specify the device I used for my measurement, I add "photometry" and "Infinite M200 plate reader (Tecan)" as the **technology type** and **technology platform**, respectively.
        </ViolaSays>

    </Steps>
</Card>


## Separate different assay elements

- reusablity

import IMG2 from "@images/start-here/arc-prototypic-assay-modular2.svg"

<img src={IMG2.src} style="width:80%;display: block; margin: 20px" />


### Add assay data and protocols


<Card icon="pen" title="ARCitect">
  <Steps>

  1. Right-click on the `dataset` folder and select `Import Files`.
  2. Select the file `sugar_result.csv` from the demo data and click `Open`.
  3. Right-click on the `protocols` folder and select `Import Files`.
  4. Select the file `sugar_extraction_protocol.md` from the demo data and click `Open`.
  5. The files are added to your ARC.
    
        <FileTree>
        - assays
            - SugarMeasurement
                - dataset
                    - **sugar_result.csv**
                - protocols
                    - **sugar_extraction_protocol.md**
                - isa.assay.xlsx
                - README.md
        - ...
        </FileTree>

  </Steps>
</Card>


### Isolate the lab processes in an assay

import IMG from "@images/start-here/arc-prototypic-assay-modular1.svg"

<img src={IMG.src} style="width:80%;display: block; margin: 20px" />

### Add one annotation table for each lab process

<Card icon="pen" title="ARCitect">
    <Steps>

    1. Click on the assay's name (`SugarMeasurement`) in the file tree.
    2. At the bottom of the right panel, click on the plus icon right next to the `Metadata` sheet to add one sheet for each lab process.
        
        <ViolaSays>
            I add two tables: one for "SugarExtraction" and one for "SugarMeasurement"
        </ViolaSays>

    3. After adding the tables, right-click on each sheet's tab to rename them accordingly.
    </Steps>
</Card>

### Parametric description of the lab process

#### Sugar extraction

![](@images/start-here/arc-prototypic-assay-labprocess-parameters1.svg)

#### Sugar measurement

![](@images/start-here/arc-prototypic-assay-labprocess-parameters2.svg)


### Use templates to describe the lab processes

<Card icon="pen" title="ARCitect">
    <Steps>

    1. Click on the assay's name (`SugarMeasurement`) in the file tree.
    2. Select the "SugarExtraction" table added in the previous step.
    3. In the `New Table!` widget, click on the `Add Template` button next to `Start from an existing template!`.
    4. From the `Select community` dropdown, select "Training"
    5. Click the name of the template `Training - Sugar extraction` and <btn>select</btn>

        <ViolaSays>
            This shows me that the template has an `Input`, `Output` as well as two `Parameter` columns: "Vortex Mixer" and "Temperature"
        </ViolaSays>

    6. Click <btn>Add template</btn> to add the columns to your table.


        Input [Sample Name]	| Parameter [Vortex Mixer] | Parameter [Temperature] | Output [Sample Name] 
        --- | --- | --- | --- 
            |     |     |   

    7. At the bottom of the table, type `6` in the text field and click <btn>+</btn> to add six rows to your table.
    8. Transer the sample names of your study's `Output [Sample Name]` to the `Input [Sample Name]` column

        Input [Sample Name]	| Parameter [Vortex Mixer] | Parameter [Temperature] | Output [Sample Name] 
        --- | --- | --- | --- 
        Cold1_leaf
        Cold2_leaf
        Cold3_leaf
        RT1_leaf
        RT2_leaf
        RT3_leaf

    9. Add information of the `sugar_extraction_protocol.md` into the table, e.g. 


        Input [Sample Name]	| Parameter [Vortex Mixer] | Parameter [Temperature] | Output [Sample Name] 
        --- | --- | --- | --- 
        Cold1_leaf  | 3 seconds | 95 degree celsius | 
        Cold2_leaf  | 3 seconds | 95 degree celsius | 
        Cold3_leaf  | 3 seconds | 95 degree celsius | 
        RT1_leaf    | 3 seconds | 95 degree celsius | 
        RT2_leaf    | 3 seconds | 95 degree celsius | 
        RT3_leaf    | 3 seconds | 95 degree celsius | 

    10. Fill the `Output [Sample Name]` column: 
        - Select the six cells from `Input`, and copy the sample names (right-click -> Copy)
        - Select the cell below `Output [Sample Name]` and paste the sample names (right-click -> Paste all)
        - Click Right-click -> Update on a cell below `Output [Sample Name]` to change the sample names
        - Type "leaf" in the `Regex` and "sugar-ext" in the `Replacement` fields
        - Click <btn>Submit</btn>

        Input [Sample Name]	| Parameter [Vortex Mixer] | Parameter [Temperature] | Output [Sample Name] 
        --- | --- | --- | --- 
        Cold1_leaf  | 3 seconds | 95 degree celsius | Cold1_sugar-ext
        Cold2_leaf  | 3 seconds | 95 degree celsius | Cold2_sugar-ext
        Cold3_leaf  | 3 seconds | 95 degree celsius | Cold3_sugar-ext
        RT1_leaf    | 3 seconds | 95 degree celsius | RT1_sugar-ext
        RT2_leaf    | 3 seconds | 95 degree celsius | RT2_sugar-ext
        RT3_leaf    | 3 seconds | 95 degree celsius | RT3_sugar-ext  

    </Steps>
</Card>


<Card icon="pen" title="ARCitect">

Follow the same steps to fill the "Sugar Measurement" table

    <Steps>
    1. Select the "SugarMeasurement" table.
    2. Load the template `Training - Sugar measurement` 
    3. Add the ` Output [Sample Name]` values (*_sugar-ext) of the "SugarExtraction" to the `Input [Sample name]` column
    4. Fill the parameter columns
        - Parameter [`technical replicate`] of 1,2,3,1,2,3
        - Parameter [`sample volume`] of 10 `microliter`
        - Parameter [`buffer volume`] of 190 `microliter`
        - Parameter [`measurement device`] Infinite M200 plate reader (Tecan)
        - Parameter [`cycle count`] of 5
    5. Use the `File Picker` feature to import the results of sugar measuremnt (`sugar_result.csv`) into `Output [Data]` 
    </Steps>
</Card>


---

## Assay 2: Proteomics Assays

<ViolaSays>
    Alright, this all makes sense. However, it seems a bit tedious, if I want to achieve this in detail for more complex assays. And besides, I'm honestly not an expert of proteomics data and I am not sure I can fill out all information about my proteomics assay. 
</ViolaSays>

![](@images/start-here/arc-prototypic-assay-sop1.svg)

### Save time using standard methods and SOPs

![](@images/start-here/arc-prototypic-assay-sop2.svg)
![](@images/start-here/arc-prototypic-assay-sop3.svg)

### Apply standard procedures to sample records

<Card icon="pen" title="ARCitect">
    <Steps>

    1. add `Proteomics_MS` assay (NAMING!)
        - import raw data folder (./MS_Raw/) into dataset
        - import local assay template: AssayTemplate_Proteomics_MS.json (formatted as ARCtrl json)
        - fill out the two tables (tables might be in wrong order!)

    2. add `Proteomics_DataAnalysis` assay
        - import dummy.fasta into dataset
        - import proteomics result (./MSFraggerOutput/combined_protein.tsv.csv) into dataset
        - import local assay template (./AssayTemplate_Proteomics_DataAnalysis.json, formatted as ARCtrl json)
        - fill out table

    </Steps>
</Card>



----
----


## ARC builds on ISA to link data

  - Samples are linked study-to-assay, assay-to-assay
  - Raw data is linked to assays
  - Protocols can be referenced
  - ...

![](@images/core-concepts/isa-files-types/isa-files02.svg)

## Link samples across studies and assays

1. Use the `Output [Sample Name]` of studies/talinum_drought/isa.study.xlsx as the `Input [Sample Name]` to **rna-extraction**.
1. Use the `Output [Sample Name]` of **rna-extraction** as the `Input [Sample Name]` to **illumina-libraries**.

**Seeds** &mdash;*Plant growth*&#x21FE; **Leaves** &mdash;*RNA Extraction*&#x21FE; **RNA** &mdash;*Illumina*&#x21FE; **fastq files**

## Link dataset files to samples

1. In the *Building Blocks* widget select *Output* -> *Raw Data File*.
2. Click <kbd>Add Column</kbd>.

:bulb: You see a warning about a changed output column.

3. Click <kbd>Continue</kbd>.
4. Go to the *File Picker* tab and click <kbd>Pick file names</kbd>.
5. Select and open the *fastq.gz files from the dataset folder.
6. Copy / paste them to the `Raw Data File`. 

:bulb: This allows you to link your samples to the resulting raw data files.
